### PR TITLE
Prepare 1.2.1 release

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Prepare the 1.2.1 bugfix release.

### Details and comments

On hold as there are some PRs still missing.

### CODE REVIEW

**Review by ElePT — DISMISSED** [review]
LGTM

**Comment by Cryoris** [comment]
There are still some PRs incoming 🙂

File: releasenotes/notes/1.2/fix-2q-basis-decomposer-non-std-kak-gate-edc69ffb5d9ef302.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+fixes:
+  - |
+    Fixed a bug in :class:`.TwoQubitBasisDecomposer` where the Rust-based code
+    would panic if the given KAK gate was not a default version of the standard gates.
+    (for example, using ``CXGate(ctrl_state=0)``).
```
Comment (ElePT) [inline releasenotes/notes/1.2/fix-2q-basis-decomposer-non-std-kak-gate-edc69ffb5d9ef302.yaml]:
```suggestion
    would panic if the given KAK gate was not a default version of the standard gates (for example, using ``CXGate(ctrl_state=0)``).
```

**Review by ElePT — COMMENTED** [review]
RLGMT, just one small suggestion

**Review by ElePT — APPROVED** [review]

**Pull request merged**